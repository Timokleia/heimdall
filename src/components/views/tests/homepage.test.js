import React from "react";
import Homepage from "../homepage";
import Enzyme, { mount, render, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { MemoryRouter, Route } from "react-router-dom";

Enzyme.configure({ adapter: new Adapter() });
describe("Homepage", () => {
  it("Renders logged out", () => {
    const wrapper = shallow(
      <MemoryRouter
        initialEntries={[{ pathname: "/160677/1", key: "testKey" }]}
      >
        <Route key='testRouter' path='/:system_id/:event_id'>
          <Homepage />
        </Route>
      </MemoryRouter>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders logged in on event level", () => {
    localStorage.setItem(
      "accessToken",
      JSON.stringify({ username: "myusername" })
    );
    const wrapper = shallow(
      <MemoryRouter
        initialEntries={[{ pathname: "/160677/1", key: "testKey" }]}
      >
        <Route key='testRouter' path='/:system_id/:event_id'>
          <Homepage />
        </Route>
      </MemoryRouter>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
