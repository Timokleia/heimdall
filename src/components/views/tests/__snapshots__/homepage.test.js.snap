// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage Renders logged in on event level 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/160677/1",
          "search": "",
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "testKey",
        "pathname": "/160677/1",
        "search": "",
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Route
    key="testRouter"
    path="/:system_id/:event_id"
  >
    <Homepage />
  </Route>
</Router>
`;

exports[`Homepage Renders logged out 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/160677/1",
          "search": "",
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "testKey",
        "pathname": "/160677/1",
        "search": "",
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Route
    key="testRouter"
    path="/:system_id/:event_id"
  >
    <Homepage />
  </Route>
</Router>
`;
